File: app/src/main/java/com/example/djikstra/Node.java
package com.example.djikstra;

import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@RequiredArgsConstructor
public class Node implements Comparable<Node> {
    private final String name;
    private Integer distance = Integer.MAX_VALUE;
    private List<Node> shortestPath = new LinkedList<>();
    private Map<Node, Integer> adjacentNodes = new HashMap<>();

    public void addAdjacentNode(Node node, int weight) {
        adjacentNodes.put(node, weight);
    }
    @Override
    public int compareTo(Node node){
        return Integer.compare(this.distance, node.getDistance());
    }
    public void main(String[] args){
    Node nodeA = new Node("a");
    Node nodeB = new Node("a");
    Node nodeC = new Node("a");
    Node nodeD = new Node("a");
    Node nodeE = new Node("a");
    Node nodeF = new Node("a");

        nodeA.addAdjacentNode(nodeB, 2);
        nodeA.addAdjacentNode(nodeC, 4);

        nodeB.addAdjacentNode(nodeC, 3);
        nodeB.addAdjacentNode(nodeD, 1);
        nodeB.addAdjacentNode(nodeE, 5);

        nodeC.addAdjacentNode(nodeD, 2);

        nodeD.addAdjacentNode(nodeE, 1);
        nodeD.addAdjacentNode(nodeF, 4);

        nodeE.addAdjacentNode(nodeF, 2);
    }
}

File: app/src/main/java/com/example/djikstra/node.kt
package com.example.djikstra

import lombok.Getter
import lombok.RequiredArgsConstructor
import lombok.Setter
import org.w3c.dom.Node
import java.util.*


@RequiredArgsConstructor
abstract class node {

    abstract var name: String
    private var distance: Int = Int.MAX_VALUE
    get() = distance
   // private var shortestPath:

    private var adjacentNodes = HashMap<node, Int>()

    fun  addAdjacentNode(node: node,weight:Int){
        adjacentNodes.put(node,weight)
    }
    fun deleteAdjacentNode(node: node,weight: Int){
        adjacentNodes.remove(node,weight)
    }
    @Override
    open fun compareTo(node: node): Int {
        return Integer.compare(distance, node.distance)
    }


}